[Workflow] Do to talk about workflow, and explain what support option is for (#20112)
symfony/symfony-docs: workflow/workflow-and-state-machine.rst

                    marking_store:
                         type: 'method'
                         property: 'currentPlace'

>                     # The supports options is useful only if you are using twig functions ('workflow_*')

Reviewer: ```suggestion
                    # the "supports" option is useful only if you are using Twig functions ('workflow_*')
```

Reviewer 2: I did these changes while merging. Thanks!

                        <framework:property>currentPlace</framework:property>
                    </framework:marking-store>

>                     <!-- The supports options is useful only if you are using twig functions ('workflow_*') -->

Reviewer: ```suggestion
                    <!-- The "support" option is useful only if you are using Twig functions ('workflow_*') -->
```

            $pullRequest
                ->type('state_machine')

>                 // The supports options is useful only if you are using twig functions ('workflow_*')

Reviewer: ```suggestion
                // The "supports" option is useful only if you are using Twig functions ('workflow_*')
```
